lib/views/movie_list_view.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+
+import '../repositories/movie_repository.dart';
+import '../text_styles/custom_text_style.dart';
+import '../ui_constants.dart';
+import 'movie_details_view.dart';
+import '../entities/movie.dart';
+
+class MovieListView extends StatefulWidget {
+  const MovieListView({super.key});
+
+  @override
+  State<MovieListView> createState() => _MovieListViewState();
+}
+
+class _MovieListViewState extends State<MovieListView> {
+  final MovieRepository movieRepository = MovieRepository();
+  Future<List<Movie>>? movies;
+  static const double dividerHeight = 20;
+  static const String errorMessage = 'An error has occurred while loading data';
+
+  @override
+  void initState() {
+    super.initState();
+    movies = movieRepository.getMovies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      builder: (BuildContext context, AsyncSnapshot<List<Movie>> snapshot) {
+        if (snapshot.hasData) {
+          List<Movie>? movies = snapshot.data;
+          return ListView.separated(
+            itemCount: movies!.length,
+            itemBuilder: (context, index) {
+              return GestureDetector(
+                onTap: () {
+                  Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                      builder: (context) => MovieDetailsVIew(
+                        movie: movies[index],
+                      ),
+                    ),
+                  );
+                },
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    Text(
+                      movies[index].title,
+                      textAlign: TextAlign.center,
+                      style: CustomTextStyle.overviewTextStyleBoldWhite,
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.all(
+                          MovieDetailsUiConstants.releaseDateRowPadding),
+                      child: Image.network(
+                        '${MovieDetailsUiConstants.imageBaseUrl}${movies[index].posterPath}',
+                      ),
+                    ),
+                  ],
+                ),
+              );
```

Comment:
You could extract this as a standalone widget, improving readabilit

Reply:
done